test/unit/smt_util/node_traversal_black.h

```diff
@@ -0,0 +1,150 @@
+/*********************                                                        */
+/*! \file node_traversal_black.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Alex Ozdemir
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Black box testing of note traversal iterators.
+ **/
+
+#include <cxxtest/TestSuite.h>
+
+// Used in some of the tests
+#include <algorithm>
+#include <cstddef>
+#include <iterator>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "expr/expr_manager.h"
+#include "expr/node.h"
+#include "expr/node_builder.h"
+#include "expr/node_manager.h"
+#include "expr/node_value.h"
+#include "smt_util/node_traversal.h"
+
+using namespace CVC4;
+using namespace CVC4::kind;
+using namespace std;
+
+class NodeBlack : public CxxTest::TestSuite
+{
+ private:
+  NodeManager* d_nodeManager;
+  NodeManagerScope* d_scope;
+
+ public:
+  void setUp() override
+  {
+    d_nodeManager = new NodeManager(NULL);
+    d_scope = new NodeManagerScope(d_nodeManager);
+  }
+
+  void tearDown() override
+  {
+    delete d_scope;
+    delete d_nodeManager;
+  }
+
+  void testPreincrementIteration()
+  {
+    Node tb = d_nodeManager->mkConst(true);
+    Node eb = d_nodeManager->mkConst(false);
+    Node cnd = d_nodeManager->mkNode(XOR, tb, eb);
+
+    auto traversal = NodePostorderIterable(cnd);
+    NodePostorderIterator i = traversal.begin();
+    NodePostorderIterator end = traversal.end();
+    TS_ASSERT(*i == tb);
+    TS_ASSERT(i != end);
+    ++i;
+    TS_ASSERT(*i == eb);
+    TS_ASSERT(i != end);
+    ++i;
+    TS_ASSERT(*i == cnd);
+    TS_ASSERT(i != end);
+    ++i;
+    TS_ASSERT(i == end);
```

Comment:
```suggestion
    TS_ASSERT_EQUALS(*i, tb);
    TS_ASSERT_NOT_EQUALS(i, end);
    ++i;
    TS_ASSERT_EQUALS(*i, eb);
    TS_ASSERT_NOT_EQUALS(i, end);
    ++i;
    TS_ASSERT_EQUALS(*i, cnd);
    TS_ASSERT_NOT_EQUALS(i, end);
    ++i;
    TS_ASSERT_EQUALS(i, end);
```

Reply:
Done in 0a0781f

Reply:
Actually, I used `TS_ASSERT_DIFFERS` in place of `TS_ASSERT_NOT_EQUALS`. I don't believe the latter is defined.